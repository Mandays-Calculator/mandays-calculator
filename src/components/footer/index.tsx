export { default as Footer } from "./Footer";
export { StyledFooterContainer, StyledFooterText } from "./styles";
