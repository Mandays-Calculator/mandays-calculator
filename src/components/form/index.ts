export * from "./checkbox";
export * from "./checkbox-group";
export * from "./text-field";
export * from "./date-picker";
