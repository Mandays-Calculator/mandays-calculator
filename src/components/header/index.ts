export {
  StyledAppBar,
  StyledToolBarContainer,
  StyledSearchSelect,
  popperStyle,
  StyledProjectTitle,
} from "./styles";
