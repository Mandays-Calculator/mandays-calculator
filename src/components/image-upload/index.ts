export { default as ImageUpload } from "./ImageUpload";
export * from "./styles";
