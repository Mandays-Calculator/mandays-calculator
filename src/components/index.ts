export * from "./form/text-field";
export * from "./form/Form";
export * from "./title";
