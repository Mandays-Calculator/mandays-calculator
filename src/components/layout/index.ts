export { default as Layout } from "./Layout";
export {
  StyledLayoutContainer,
  StyledLayoutContent,
  ContentContainer,
} from "./styles";
