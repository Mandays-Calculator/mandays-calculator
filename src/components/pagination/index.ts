export { default as Pagination } from "./Pagination";
export * from "./styles";
