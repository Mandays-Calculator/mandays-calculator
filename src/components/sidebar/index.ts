export {
  DrawerHeader,
  StyledItemText,
  StyledListItem,
  StyledListItemIcon,
  StyledDrawer,
  StyledList,
} from "./styles";

export { default as Sidebar } from "./Sidebar";
