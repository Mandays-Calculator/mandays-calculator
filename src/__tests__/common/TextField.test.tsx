import { render } from "@testing-library/react";
import { TextField } from "~/components";

describe("TextField Component", () => {
  it("renders without crashing", () => {
    const { getByTestId, asFragment } = render(<TextField name="test-textfield" />);
    const textField = getByTestId("text-field-component");
    expect(textField).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with label", () => {
    const { getByTestId, asFragment } = render(
      <TextField
        name="text-field-with-labe"
        label="sample label"
      />
    );

    const textField = getByTestId("text-field-component");
    expect(textField).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
