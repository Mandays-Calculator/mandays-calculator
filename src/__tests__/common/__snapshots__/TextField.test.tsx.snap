// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component renders with label 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-hwnj0i-MuiStack-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      sample label
    </p>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="text-field-component"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="text-field-with-labe"
          placeholder="Please Input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextField Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-hwnj0i-MuiStack-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      data-testid="text-field-component"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="test-textfield"
          placeholder="Please Input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
